### PR TITLE
fix: All .NET functions are async now; and need to be awaited.

### PR DESCRIPTION
**Pull request by JamieSinn**
Also reformat according to .NET standards.

### CODE REVIEW

File: proxies/dotnet/Controllers/Location.cs [hunk 1]
```diff
@@ -142,96 +154,127 @@ public async Task<object> Post([FromBody] PostBody body, string location)
...
+
         return result;
     }
 
-    private async Task<CommandResult> InvokeCommand(object entity, string command, bool isAsync, List<object> parsedParams) {
+    private async Task<CommandResult> InvokeCommand(object entity, string command, bool isAsync,
+        List<object> parsedParams)
+    {
         var parsedCommand = char.ToUpper(command[0]) + command.Substring(1);
-        parsedCommand = entity is DevCycleCloudClient ? parsedCommand + "Async" : parsedCommand;
```
Comment (JamieSinn) [inline proxies/dotnet/Controllers/Location.cs:188]: We removed the suffix here.

File: proxies/dotnet/Controllers/Location.cs [hunk 2]
```diff
@@ -142,96 +154,127 @@ public async Task<object> Post([FromBody] PostBody body, string location)
...
             parsedParams[0] = customDataDict;
         }
 
         object result = null;
-        if (isAsync) {
-            dynamic? task = commandMethod?.Invoke(entity, parsedParams.ToArray());
-            result = task == null ? result : (await task);
-        } else {
-            result = commandMethod?.Invoke(entity, parsedParams.ToArray()) ?? result;
-        }
```
Comment (JamieSinn) [inline proxies/dotnet/Controllers/Location.cs:213]: All functions return a task; and need to be awaited.

File: proxies/dotnet/dotnet.csproj [hunk 3]
```diff
@@ -8,8 +8,8 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="7.0.1" />
-    <PackageReference Include="DevCycle.SDK.Server.Cloud" Version="1.6.0" />
-    <PackageReference Include="DevCycle.SDK.Server.Local" Version="2.3.0" />
+    <PackageReference Include="DevCycle.SDK.Server.Cloud" Version="3.0.1" />
+    <PackageReference Include="DevCycle.SDK.Server.Local" Version="4.0.1" />
```
Comment (JamieSinn) [inline proxies/dotnet/dotnet.csproj:12]: Bumped to the latest

**Review by chris-hoefgen — APPROVED** [review]
LGTM. What are you using for code formatting here?

**Comment by JamieSinn** [comment]
Rider

**Pull request merged**